src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -58,24 +59,24 @@ public sealed class Arguments : TransformInputBase
         internal const string ShortName = "Onnx";
         internal const string LoaderSignature = "OnnxTransform";
 
-        public readonly string Input;
-        public readonly string Output;
-        public readonly ColumnType OutputType;
+        public readonly string[] Inputs;
+        public readonly string[] Outputs;
+        public readonly ColumnType[] OutputTypes;
 
         private static VersionInfo GetVersionInfo()
         {
             return new VersionInfo(
                 modelSignature: "ONNXSCOR",
-                verWrittenCur: 0x00010001, // Initial
-                verReadableCur: 0x00010001,
+                verWrittenCur: 0x00010002, // Initial
+                verReadableCur: 0x00010002,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
 	        loaderAssemblyName: typeof(OnnxTransform).Assembly.FullName);
         }
 
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string inputColumn, string outputColumn)
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string[] inputColumns, string[] outputColumns)
```

Comment:
>public [](start = 8, length = 6)

can it be internal?

Reply:
TF is same way  (see below)-- I assume this is required to be public due to different ways it can be instantiated.

[https://github.com/dotnet/machinelearning/blob/cb9effcd091c60fa291aad96cc18c14ddf841b6f/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L195](https://github.com/dotnet/machinelearning/blob/cb9effcd091c60fa291aad96cc18c14ddf841b6f/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L195)


---
In reply to: [233267511](https://github.com/dotnet/machinelearning/pull/1586#discussion_r233267511) [](ancestors = 233267511)